src/vmm/src/devices/virtio/vhost_user.rs

```diff
@@ -442,7 +443,17 @@ impl<T: VhostUserHandleBackend> VhostUserHandleImpl<T> {
             // No matter the queue, we set irq_evt for signaling the guest that buffers were
             // consumed.
             self.vu
-                .set_vring_call(*queue_index, &irq_trigger.irq_evt)
+                .set_vring_call(
+                    *queue_index,
+                    interrupt
+                        .notifier(VirtioInterruptType::Queue(
+                            (*queue_index).try_into().unwrap_or_else(|_| {
+                                panic!("vhost-user: invalid queue index: {}", *queue_index)
```

Comment:
Same here, please fix this insane conversion code.

Reply:
same as before, I think that the solution is fixing the queue indexes at their definitions. If we need to pass them somewhere that expects a `u16` at _some_ point we will need to do a conversion.